testutil/otel.go

```diff
@@ -0,0 +1,149 @@
+package testutil
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/utility"
+	"github.com/mongodb/grip"
+	"github.com/pkg/errors"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	"go.opentelemetry.io/otel/sdk/resource"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.17.0"
+	"go.opentelemetry.io/otel/trace"
+)
+
+var (
+	packageName                 = fmt.Sprintf("%s%s", evergreen.PackageName, "/testutil")
+	serviceName                 = "evergreen-tests"
+	otelCollectorEndpointEnvVar = "OTEL_COLLECTOR_ENDPOINT"
+	otelTraceIDEnvVar           = "OTEL_TRACE_ID"
+	otelParentIDEnvVar          = "OTEL_PARENT_ID"
+)
+
+func TestSpan(ctx context.Context, t *testing.T) context.Context {
+	if !trace.SpanContextFromContext(ctx).IsValid() {
+		return spanForRootTest(ctx, t)
+	} else {
+		return spanForChildTest(ctx, t)
+	}
+}
+
+func spanForRootTest(ctx context.Context, t *testing.T) context.Context {
+	collectorEndpoint := os.Getenv(otelCollectorEndpointEnvVar)
+	traceIDString := os.Getenv(otelTraceIDEnvVar)
+	spanIDString := os.Getenv(otelParentIDEnvVar)
+	if collectorEndpoint == "" || traceIDString == "" || spanIDString == "" {
+		return ctx
```

Comment:
No error message?

Reply:
This condition is like if someone is running the test locally so these variables won't be set in the environment

Reply:
Okay, cool.

Reply:
done